src/test/java/bio/terra/service/auth/iam/sam/SamIamTest.java

```diff
@@ -525,242 +296,493 @@ public void testCreateSnapshotResourceRequestWithPolicySpecifications() throws A
-        samIam.createGroup(accessToken, groupName),
-        equalTo(groupEmail));
-    verify(samGroupApi).postGroup(groupName, null);
-    verify(samGroupApi).getGroup(groupName);
-  }
+      AccessPolicyMembershipV2 custodian = req.getPolicies().get(IamRole.CUSTODIAN.toString());
+      assertThat(custodian.getRoles(), contains(IamRole.CUSTODIAN.toString()));
+      assertThat(custodian.getMemberEmails(), empty());
 
-  @Test
-  public void testCreateGroupWithCreationFailure() throws ApiException {
-    String accessToken = TEST_USER.getToken();
-    String groupName = "firecloud_group_name";
-
-    ApiException samEx =
-        new ApiException(HttpStatusCodes.STATUS_CODE_CONFLICT, "Group already exists");
-    doThrow(samEx).when(samGroupApi).postGroup(groupName, null);
-    assertThrows(
-        "IamConflictException is thrown when the group already exists",
-        IamConflictException.class,
-        () -> samIam.createGroup(accessToken, groupName));
-    verify(samGroupApi).postGroup(groupName, null);
-    verify(samGroupApi, never()).getGroup(groupName);
+      AccessPolicyMembershipV2 snapshotCreator =
+          req.getPolicies().get(IamRole.SNAPSHOT_CREATOR.toString());
+      assertThat(snapshotCreator.getRoles(), contains(IamRole.SNAPSHOT_CREATOR.toString()));
+      assertThat(snapshotCreator.getMemberEmails(), empty());
+    }
   }
 
-  @Test
-  public void testCreateGroupWithEmailFetchFailure() throws ApiException {
-    String accessToken = TEST_USER.getToken();
-    String groupName = "firecloud_group_name";
-
-    ApiException samEx = new ApiException(HttpStatusCodes.STATUS_CODE_NOT_FOUND, "Group not found");
-    when(samGroupApi.getGroup(groupName)).thenThrow(samEx);
-    assertThrows(
-        "IamNotFoundException is thrown when the user cannot access their created group",
-        IamNotFoundException.class,
-        () -> samIam.createGroup(accessToken, groupName));
-    verify(samGroupApi).postGroup(groupName, null);
-    verify(samGroupApi).getGroup(groupName);
-  }
+  @Nested
+  class TestSamWithMockedConfig {
 
-  @Test
-  public void testOverwriteGroupPolicyEmails() throws InterruptedException, ApiException {
-    String accessToken = TEST_USER.getToken();
-    String groupName = "firecloud_group_name";
-    String policyName = IamRole.MEMBER.toString();
-    List<String> emails = List.of("user@a.com");
-
-    samIam.overwriteGroupPolicyEmails(accessToken, groupName, policyName, emails);
-    verify(samGroupApi).overwriteGroupPolicyEmails(groupName, policyName, emails);
-  }
+    @BeforeEach
+    void setUp() throws Exception {
+      when(configurationService.getParameterValue(ConfigEnum.SAM_RETRY_MAXIMUM_WAIT_SECONDS))
+          .thenReturn(0);
+      when(configurationService.getParameterValue(ConfigEnum.SAM_RETRY_INITIAL_WAIT_SECONDS))
+          .thenReturn(0);
+      when(configurationService.getParameterValue(ConfigEnum.SAM_OPERATION_TIMEOUT_SECONDS))
+          .thenReturn(0);
+    }
```

Comment:
If you change `configurationService` so it's no longer mocked then this initialization is no longer needed (since the default values for these properties are all `0` already) and then I don't think you need the `TestSamWithMockedConfig` nested class anymore.